code/game/objects/effects/step_triggers.dm

```diff
@@ -226,3 +226,21 @@
 	name = "a grave mistake";
 	sound = 'sound/effects/hallucinations/i_see_you1.ogg'
 	triggerer_only = 1
+
+/// Forces a given outfit onto any carbon which crosses it, for event maps
+/obj/effect/step_trigger/outfitter
+	mobs_only = TRUE
+	///outfit to equip
+	var/datum/outfit/dresscode
```

Comment:
does this actually supply enough control?

Reply:
For my purposes. I fail to see any other predictably applicable control to add. I am however changing the var name to be less obtuse for mappers.